src/mlpack/methods/neighbor_search/ns_model.hpp

```diff
@@ -37,6 +49,127 @@ struct NSModelName<FurthestNeighborSort>
   static const std::string Name() { return "furthest_neighbor_search_model"; }
 };
 
+class MonoSearchVisitor : public boost::static_visitor<void>
```

Comment:
Ah, I hate to be picky, so I hope I am not giving too much work, but do you think you could add documentation for these classes and their members and member functions?  I am not sure everyone who looks through this file will be familiar with the visitor paradigm; there is no need to explain what that is in your comments, but it may be useful to have comments along the lines of "MonoSearchVisitor executes a monochromatic neighbor search on the given NSType", or something like this.  Maybe a few more words are useful to explain that better. :)